lazy_dataset/database_cli.py

```diff
@@ -0,0 +1,241 @@
+
+            def _enumerate(self, seq):
+                """like enumerate, but with an upper limit on the number of items"""
+                length = 1
+                overlength = 0
+                for idx, x in enumerate(seq):
+                    length += 1
+                    if self.max_seq_length and idx >= self.max_seq_length:
+                        overlength += 1
+                        # self.text(',')
+                        # self.breakable()
+                        # self.text('...')
+                        # return
+                    else:
+                        yield idx, x
+                if overlength:
+                    self.text(',')
+                    self.breakable()
+                    self.text(f'...,  # {length} - {length-overlength}')
+                    p.break_()
+
+        p = RepresentationPrinter(
+            sys.stdout, max_width=max_width, max_seq_length=s)
+
+        indent = 4
+
+        @contextlib.contextmanager
+        def group_break(p, indent=0, open='', close=''):
+            with p.group(indent, open):
+                yield
+            p.break_()
+            if close:
+                p.text(close)
+
+        with group_break(p, 4, '{', '}'):
+            for k, v in data.items():
+                p.break_()
+                if k == 'alias':
+                    k = f'{k!r}: '
+                    with p.group(len(k), k):
+                        p.pretty(v)
+                        p.text(',')
+                elif k != 'datasets':
+                    p.text(f'{k!r}: ...,')
+                else:
+                    with group_break(p, indent, f'{k!r}: {{', '},'):
+                        for i2, (dataset_name, dataset) in enumerate(v.items()):
+                            p.break_()
+                            if i2 >= d:
+                                p.text(f'{dataset_name!r}: ...,  # {len(dataset)} examples')
+                                continue
+                            with group_break(p, indent, f'{dataset_name!r}: {{', '},'):
+                                for i3, (example_id, example) in enumerate(dataset.items()):
+                                    if i3 >= n:
+                                        break
+                                    p.break_()
+                                    with group_break(p, indent, f'{example_id!r}: {{', '},'):
+                                        for key, value in example.items():
+                                            p.break_()
+                                            # p.text(f'{key}: ')
+                                            key = f'{key!r}: '
+                                            with p.group(len(key), key):
+                                                p.pretty(value)
+                                                p.text(f',')
+                                p.break_()
+                                p.text(f'...,  # {len(dataset)} - {n} examples')
```

Comment:
Only print this when not all examples are printed? Again, I would change this to something easier to understand like `skipped {len(dataset) - n} examples of {len(dataset)}` or `showing {n} examples of {len(dataset)}`

Reply:
Will change to the `skipped` variant

Reply:
Done